spiped/main.c

```diff
@@ -274,50 +284,92 @@ main(int argc, char * argv[])
 			warnp_syslog(1);
 	}
 
+	/* Resolve bind address. */
+	if (opt_b) {
+		char * a = strchr(opt_b, ':');
+		char * b = strrchr(opt_b, ':');
+		char * c = strrchr(opt_b, ']');
+		/* If there is no ':', it's ipv4 address (or Unix domain socket). */
+		if (b == NULL) {
+			if (asprintf(&bind_addr, "%s:0", opt_b) == -1) {
+			    warnp("asprintf");
+			    goto err2;
+			}
+		}
+		/* If there are two different ':', it's ipv6 address. */
+		else if (a != b) {
+			/* If there is no ']', we need to add '[]' and ':0'. */
+			if (c == NULL) {
+				if (asprintf(&bind_addr, "[%s]:0", opt_b) == -1) {
+				    warnp("asprintf");
+				    goto err2;
+				}
+			}
+			/* If there is nothing after ']', we need to add ':0'. */
```

Comment:
If the ipv6 address already contains a port number, then `bind_addr` is unset.
(I've fixed this in #358)